src/services/utility.service.ts

```diff
@@ -54,37 +57,30 @@ class UtilityService {
    * @param inputText Input text for search
    * @param setSearchResults Function for setting state of search results
    * @param setDisplayValue Function for setting display text of search
+   *
+   * TODO: Add current location as a searchable element by passing it as
+   * a parameter and including it in the search.
    */
   public updateSearchResults = (
     inputText: string,
-    setSearchResults: (locations: (POI | Building)[]) => void,
-    setDisplayValue: (text: string) => void
+    setSearchResults: (locations: SearchResult[]) => void,
+    setLocationDisplay: (display: string) => void
   ) => {
-    const locationsToSearch: (POI | Building)[] = [...POIInfo, ...Buildings];
-    const locations: (POI | Building)[] = locationsToSearch.filter(
+    const locations: SearchResult[] = this.locationsToSearch.filter(
       (location) => {
-        if (isPOI(location)) {
-          return (
-            location.displayName
-              .toUpperCase()
-              .search(inputText.toUpperCase()) !== -1
-          );
-        }
         return (
-          location.displayName.toUpperCase().search(inputText.toUpperCase()) !==
-            -1 ||
-          BuildingId[location.id]
-            .toUpperCase()
-            .search(inputText.toUpperCase()) !== -1
+          location.searchString
+            .toLowerCase()
+            .search(inputText.toLowerCase()) !== -1
         );
       }
     );
 
-    const narrowedLocations: (POI | Building)[] =
+    const narrowedLocations: SearchResult[] =
```

Comment:
My only concern would be the coupling of types. Is there a particular reason?

Reply:
The reason is to avoid type checking, which was a really bad code smell in our code previously.

these types need to be brought together (coupled, as you say) because they are used in the same context

Reply:
I'm sorry, I still don't understand.
I'm not sure what POI and Building are at the moment. Building is it's own interface and POI is a marker location? StartLocation is a MarkerLocation? If so I'm not sure if the contract between pathfinding service and omnibus is possible anymore.

Reply:
I'm not sure what you're reffering to, we can discuss about this at another time.